Mathlib/Analysis/InnerProductSpace/Basic.lean

```diff
@@ -113,8 +115,25 @@ class InnerProductSpace (𝕜 : Type*) (E : Type*) [RCLike 𝕜] [NormedAddCommG
   add_left : ∀ x y z, inner (x + y) z = inner x z + inner y z
   /-- The inner product is conjugate linear in the first coordinate. -/
   smul_left : ∀ x y r, inner (r • x) y = conj r * inner x y
+
+/-- An inner product space is a vector space with an additional operation called inner product.
+The norm could be derived from the inner product, instead we require the existence of a norm and
+the fact that `‖x‖^2 = re ⟪x, x⟫` to be able to put instances on `𝕂` or product
+spaces.
+
+To construct a norm from an inner product, see `InnerProductSpace.ofCore`.
+-/
+class InnerProductSpace (𝕜 : Type*) (E : Type*) [RCLike 𝕜] [NormedAddCommGroup E] extends
```

Comment:
Can you explain why you want to have two different classes instead of just one named `InnerProductSpace` but only assuming `[RCLike 𝕜] [SeminormedAddCommGroup E]`? Then, depending on whether you want to have a norm or a seminorm, you would assume `[SeminormedAddCommGroup E]` or `[NormedAddCommGroup E]`, but use the same class `InnerProductSpace 𝕜 E`.

Reply:
sorry, I misunderstood the discussion above. As it was suggested to add `InnerProductSpace.Core` and `PreInnerProductSpace.Core`, I thought it would apply to the class `Pre/InnerProductSpace` as well. I removerd `PreInnerProductSpace`.

The theorems are split in sections without and with `[NormedAddCommGroup E]`. I wonder whether this is the best way.

Reply:
now for `InnerProductSpace.ofMatrix` in `Mathlib.LinearAlgebra.Matrix.PosDef`, I had to weaken the definition (it is `InnerProductSpace` only with `SeminormedAddCommGroup`, not `NormedAddCommGroup`. Is this fine?

Reply:
What you've done there is correct.